src/Symfony/Component/ExpressionLanguage/Tests/ExpressionLanguageTest.php

```diff
@@ -78,6 +80,27 @@ public function testConstantFunction()
         $this->assertEquals('\constant("PHP_VERSION")', $expressionLanguage->compile('constant("PHP_VERSION")'));
     }
 
+    public function testEnumFunctionWithConstantThrows()
+    {
+        $this->expectException(\TypeError::class);
+        $this->expectExceptionMessage('The string "PHP_VERSION" is not the name of a valid enum case.');
+        $expressionLanguage = new ExpressionLanguage();
+        $expressionLanguage->evaluate('enum("PHP_VERSION")');
+    }
+
+    public function testEnumFunction()
+    {
+        $expressionLanguage = new ExpressionLanguage();
+        $this->assertSame(FooEnum::Foo, $expressionLanguage->evaluate('enum("Symfony\\\\Component\\\\ExpressionLanguage\\\\Tests\\\\Fixtures\\\\FooEnum::Foo")'));
+    }
+
+    public function testBackedEnumFunction()
+    {
+        $expressionLanguage = new ExpressionLanguage();
+        $this->assertSame(FooBackedEnum::Bar, $expressionLanguage->evaluate('enum("Symfony\\\\Component\\\\ExpressionLanguage\\\\Tests\\\\Fixtures\\\\FooBackedEnum::Bar")'));
+        $this->assertSame('Foo', $expressionLanguage->evaluate('enum("Symfony\\\\Component\\\\ExpressionLanguage\\\\Tests\\\\Fixtures\\\\FooBackedEnum::Bar").value'));
```

Comment:
test cases are missing for the compiled version

Reply:
Added! 🙂